### PR TITLE
chore(TDB): Setup of Error Handlers and Serialization

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
### Description of Change:
- Added Error handlers and serializers in the server side. 
### Associated Ticket Link(s):
- https://trello.com/c/vMm9SJwS/18-error-handlers-and-serializers

### Developers Note:
- Made some minor change in the TaskPriority enum of the entity.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -3,6 +3,14 @@ import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+
+  // Enable CORS
```
Comment (mahir-mohsin-sazim) [inline src/main.ts:7]: remove comments

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 2]
```diff
@@ -3,6 +3,14 @@ import { AppModule } from './app.module';
...
+  app.enableCors({
+    origin: 'http://localhost:3001', // Your frontend URL
```
Comment (nur-zaman-sazim) [inline src/main.ts:9]: - read origin from .env
- remove comments

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 1 above)
Comment (mohammed-sharraf-sazim) [reply src/main.ts]: Comments Removed

File: src/main.ts (hunk 2 above)
Comment (mohammed-sharraf-sazim) [reply src/main.ts]: Did what you instructed

**Review by nur-zaman-sazim — APPROVED** [review]

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Pull request merged**